### ISSUE TITLE
uint64 unmarshalling

### ISSUE THREAD

**Issue by nikandfor** [gocql/gocql#629]
go uint64 type is marshallable to varint but it can't be unmarshalled if value > MaxInt64

https://github.com/gocql/gocql/blob/master/marshal.go#L439
https://github.com/gocql/gocql/blob/master/marshal.go#L418

It could be fixed somethink like this:
```
diff --git a/marshal.go b/marshal.go
index 254da30..70efaa1 100644
--- a/marshal.go
+++ b/marshal.go
@@ -401,6 +401,13 @@ func bytesToInt64(data []byte) (ret int64) {
 	return ret
 }
 
+func bytesToUint64(data []byte) (ret uint64) {
+	for i := range data {
+		ret |= uint64(data[i]) << (8 * uint(len(data)-i-1))
+	}
+	return ret
+}
+
 func unmarshalBigInt(info TypeInfo, data []byte, value interface{}) error {
 	return unmarshalIntlike(info, decBigInt(data), data, value)
 }
@@ -410,9 +417,14 @@ func unmarshalInt(info TypeInfo, data []byte, value interface{}) error {
 }
 
 func unmarshalVarint(info TypeInfo, data []byte, value interface{}) error {
-	switch value.(type) {
+	switch v := value.(type) {
 	case *big.Int:
 		return unmarshalIntlike(info, 0, data, value)
+	case *uint64:
+		if len(data) == 9 && data[0] == 0 {
+			*v = bytesToUint64(data[1:])
+			return nil
+		}
 	}
 
 	if len(data) > 8 {
diff --git a/marshal_test.go b/marshal_test.go
index d9143be..b0c31d4 100644
--- a/marshal_test.go
+++ b/marshal_test.go
@@ -675,6 +675,64 @@ func TestMarshalVarint(t *testing.T) {
 			t.Errorf("unmarshaled varint mismatch: expected %v, got %v (test #%d)", test.Unmarshaled, binder, i)
 		}
 	}
+
+	varintUint64Tests := []struct {
+		Value       interface{}
+		Marshaled   []byte
+		Unmarshaled uint64
+	}{
+		{
+			Value:       int8(0),
+			Marshaled:   []byte("\x00"),
+			Unmarshaled: 0,
+		},
+		{
+			Value:       uint8(255),
+			Marshaled:   []byte("\x00\xFF"),
+			Unmarshaled: 255,
+		},
+		{
+			Value:       big.NewInt(math.MaxInt32),
+			Marshaled:   []byte("\x7F\xFF\xFF\xFF"),
+			Unmarshaled: uint64(math.MaxInt32),
+		},
+		{
+			Value:       big.NewInt(int64(math.MaxInt32) + 1),
+			Marshaled:   []byte("\x00\x80\x00\x00\x00"),
+			Unmarshaled: uint64(int64(math.MaxInt32) + 1),
+		},
+		{
+			Value:       uint64(math.MaxInt64) + 1,
+			Marshaled:   []byte("\x00\x80\x00\x00\x00\x00\x00\x00\x00"),
+			Unmarshaled: 9223372036854775808,
+		},
+		{
+			Value:       math.MaxUint64,
+			Marshaled:   []byte("\x00\x80\x00\x00\x00\x00\x00\x00\x00"),
+			Unmarshaled: math.MaxUint64,
+		},
+	}
+
+	for i, test := range varintTests {
+		data, err := Marshal(NativeType{proto: 2, typ: TypeVarint}, test.Value)
+		if err != nil {
+			t.Errorf("error marshaling varint: %v (test #%d)", err, i)
+		}
+
+		if !bytes.Equal(test.Marshaled, data) {
+			t.Errorf("marshaled varint mismatch: expected %v, got %v (test #%d)", test.Marshaled, data, i)
+		}
+
+		binder := new(uint64)
+		err = Unmarshal(NativeType{proto: 2, typ: TypeVarint}, test.Marshaled, binder)
+		if err != nil {
+			t.Errorf("error unmarshaling varint: %v (test #%d)", err, i)
+		}
+
+		if test.Unmarshaled != binder {
+			t.Errorf("unmarshaled varint mismatch: expected %v, got %v (test #%d)", test.Unmarshaled, binder, i)
+		}
+	}
 }
 
 func equalStringSlice(leftList, rightList []string) bool {
```

**Comment by Zariel** [comment]
Looks good, want to submit a pull request?

**Comment by nikandfor** [comment]
I'll try but without compilation.
I get error when I compile fork. 
```
package github.com/nikandfor/gocql
	imports github.com/gocql/gocql/internal/lru: use of internal package not allowed
```

**Comment by Zariel** [comment]
@nikandfor I left feedback on the PR to get it to compile and pass the tests, it should work if you clone it to `$GOPATH/src/github.com/gocql/gocql`

**Comment by nikandfor** [comment]
@Zariel thank you for help.
Now I read about build tags and I could test it on my laptop :)

**Comment by Zariel** [comment]
Fixed via #630

**Comment by Zariel** [comment]
Thanks for the contribution :)
